### PR TITLE
:wheel_of_dharma: added kubernetes config and build step

### PR DESCRIPTION
**Pull request by M4tini**
- Added `kube` dir with Kubernetes config.
- Updated CircleCI to deploy Kubernetes.

### CODE REVIEW

**Comment by M4tini** [comment]
@yoan-myparcel do you think we can get rid of the two times we use `microservice` in the configs?

File: kube/configmap.template.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
```
Comment (yoan-myparcel) [inline kube/configmap.template.yaml]: Lets make the configmap and secret as .example files.

File: .circleci/config.yml [hunk 2]
```diff
@@ -47,6 +47,65 @@ jobs:
...
+            envsubst < kube/deployment.template.yaml > kube/deployment.yaml
+            envsubst < kube/service.template.yaml > kube/service.yaml
+      - run:
+          name: Deploy
+          command: |
+            kubectl apply -f kube/configmap.yaml
+            kubectl apply -f kube/secrets.yaml
+            kubectl apply -f kube/deployment.yaml
+            kubectl apply -f kube/service.yaml
+
```
Comment (yoan-myparcel) [inline .circleci/config.yml]: Lets split service.template.yaml and introduce ingress.template.yaml in order to exclude it for production: 
```suggestion
if [[ "$(env_name)" != "production" ]]; then
    kubectl apply -f kube/ingress.yaml
fi
```

File: kube/deployment.template.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      app: ${CIRCLE_PROJECT_REPONAME}
+  template:
+    metadata:
+      labels:
+        app: ${CIRCLE_PROJECT_REPONAME}
+        revision: ${CIRCLE_SHA1}
+    spec:
+      containers:
+        - name: ${CIRCLE_PROJECT_REPONAME}
+          image: myparcelcom/microservice:${IMAGE_TAG}
```
Comment (yoan-myparcel) [inline kube/deployment.template.yaml]: We can use the CIRCLE_PROJECT_REPONAME

File: kube/service.template.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+kind: Service
+apiVersion: v1
+metadata:
+  name: ${CIRCLE_PROJECT_REPONAME}-svc
```
Comment (yoan-myparcel) [inline kube/service.template.yaml]: The internal DNS hostname for this service will become `${CIRCLE_PROJECT_REPONAME}-svc.${NS}.svc.cluster.local`

File: kube/service.template.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  namespace: ${NS}
+spec:
+  selector:
+    app: ${CIRCLE_PROJECT_REPONAME}
+    revision: ${CIRCLE_SHA1}
+  ports:
+    - protocol: TCP
+      port: 443
+      targetPort: 443
```
Comment (yoan-myparcel) [inline kube/service.template.yaml]: Maybe we can start exposing port 80 for internal communication purposes.

Comment (M4tini) [reply kube/service.template.yaml]: Decided to do this if we encounter problems.

File: .circleci/config.yml [hunk 6]
```diff
@@ -45,7 +45,71 @@ jobs:
       - run:
           name: Deploy to docker
           command: |
-            ./docker/build.sh microservice develop
+            ./docker/build.sh ${CIRCLE_PROJECT_REPONAME##*-}` develop
```
Comment (yoan-myparcel) [inline .circleci/config.yml]: Forgotten `?

**Pull request merged**